core/src/test/java/org/apache/accumulo/core/file/rfile/MultiThreadedRFileTest.java

```diff
@@ -0,0 +1,387 @@
+      }
+      FileSystem fs = FileSystem.newInstance(conf);
+      Path path = new Path("file://" + rfile.toString());
+      dos = fs.create(path, true);
+      CachableBlockFile.Writer _cbw = new CachableBlockFile.Writer(dos, "gz", conf, accumuloConfiguration);
+      writer = new RFile.Writer(_cbw, 1000, 1000);
+
+      if (startDLG)
+        writer.startDefaultLocalityGroup();
+    }
+
+    public void openWriter() throws IOException {
+      openWriter(true);
+    }
+
+    public void closeWriter() throws IOException {
+      if (deepCopy) {
+        throw new IOException("Cannot open writer on a deepcopy");
+      }
+      dos.flush();
+      writer.close();
+      dos.flush();
+      dos.close();
+    }
+
+    public void openReader() throws IOException {
+      FileSystem fs = FileSystem.newInstance(conf);
+      Path path = new Path("file://" + rfile.toString());
+
+      // the caches used to obfuscate the multithreaded issues
+      CachableBlockFile.Reader _cbr = new CachableBlockFile.Reader(fs, path, conf, null, null, AccumuloConfiguration.getDefaultConfiguration());
+      reader = new RFile.Reader(_cbr);
+      iter = new ColumnFamilySkippingIterator(reader);
+
+      checkIndex(reader);
+    }
+
+    public void closeReader() throws IOException {
+      reader.close();
+    }
+
+    public void seek(Key nk) throws IOException {
+      iter.seek(new Range(nk, null), EMPTY_COL_FAMS, false);
+    }
+  }
+
+  static Key nk(String row, String cf, String cq, String cv, long ts) {
+    return new Key(row.getBytes(), cf.getBytes(), cq.getBytes(), cv.getBytes(), ts);
+  }
+
+  static Value nv(String val) {
+    return new Value(val.getBytes());
+  }
+
+  public AccumuloConfiguration conf = null;
+
+  @Test
+  public void testMultipleReaders() throws IOException {
+    final List<Throwable> threadExceptions = Collections.synchronizedList(new ArrayList<Throwable>());
+    Map<String,MutableInt> messages = new HashMap<String,MutableInt>();
+    Map<String,String> stackTrace = new HashMap<String,String>();
+
+    final TestRFile trfBase = new TestRFile(conf);
+
+    writeData(trfBase);
+
+    trfBase.openReader();
+
+    try {
+
+      validate(trfBase);
+
+      final TestRFile trfBaseCopy = trfBase.deepCopy();
+
+      validate(trfBaseCopy);
+
+      // now start up multiple RFile deepcopies
+      int maxThreads = 10;
+      String name = "MultiThreadedRFileTestThread";
+      ThreadPoolExecutor pool = new ThreadPoolExecutor(maxThreads + 1, maxThreads + 1, 5 * 60, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>(),
+          new NamingThreadFactory(name));
+      pool.allowCoreThreadTimeOut(true);
+      try {
+        Runnable runnable = new Runnable() {
+          @Override
+          public void run() {
+            try {
+              TestRFile trf = trfBase;
+              synchronized (trfBaseCopy) {
+                trf = trfBaseCopy.deepCopy();
+              }
+              validate(trf);
```

Comment:
Good point.  I had to add things to the validate to ensure it ran long enough to incur the errors.